### PR TITLE
Refactor a little in HelmReconciler.ApplyManifest

### PR DESCRIPTION
**Pull request by sword-jin**
In current implementation of `HelmReconciler.ApplyManifest`, it returns 3 args and they dont be used in most of cases.
So in this PR, we define `AppliedResult` to make it more understandable.

Nothing changed in change anything in functional.

**Please provide a description of this PR:**

### CODE REVIEW

File: operator/pkg/helmreconciler/apply.go
Comment (hanxiaop) [inline operator/pkg/helmreconciler/apply.go:49]: I think this has changed the original condition:
```
len(processedObjs) != 0 || deployedObjects > 0
```
which uses `||` instead of `&&`?

**Review by hanxiaop — APPROVED** [review]

**Comment by sword-jin** [comment]
wow, is LGTM comment required

**Comment by hanxiaop** [comment]
> wow, is LGTM comment required ping @hanxiaop

No. If you look at `Code owner review required` part, it shows `Waiting on code owner review from istio/wg-test-and-release-maintainers. `.

**Review by jwendell — APPROVED** [review]

**Pull request merged**
